src/instrumenter/instrumentation_context.ts

```diff
@@ -118,4 +140,24 @@ export class InstrumentationContext {
 
         return this.internalInvariantCheckers.get(contract) as string;
     }
+
+    addGeneralInstrumentation(...nodes: ASTNode[]): void {
+        this.generalInstrumentationNodes.push(...nodes);
+    }
+
+    addAnnotationInstrumentation(annotation: AnnotationMetaData, ...nodes: ASTNode[]): void {
+        if (!this.evaluationStatements.has(annotation)) {
+            this.evaluationStatements.set(annotation, nodes);
+        } else {
+            (this.evaluationStatements.get(annotation) as ASTNode[]).push(...nodes);
+        }
+    }
+
+    addAnnotationCheck(annotation: AnnotationMetaData, pred: Expression): void {
+        if (!this.instrumetnedCheck.has(annotation)) {
+            this.instrumetnedCheck.set(annotation, [pred]);
+        } else {
+            (this.instrumetnedCheck.get(annotation) as Expression[]).push(pred);
+        }
```

Comment:
It would be better to remove boolean inversion (`!`) in `if` condition and swap `if`/`else` actions. Smaller amount of computations and better readability.